### DISCUSSION TITLE
Garip Bölme 1984A (Codeforces)

### DISCUSSION THREAD

**Discussion by Burak2210** [sonsuzus/sonsuzus#511] [Soru Cevap] [answered]
Bir dizinin **aralığı**, dizideki maksimum değer ile minimum değer arasındaki fark olarak tanımlanır. Örneğin, \([1, 4, 2]\) dizisinin aralığı \(4 - 1 = 3\) olur.

Size uzunluğu n ≥ 3 olan bir dizi a₁, a₂, …, aₙ veriliyor. Bu dizinin **sıralı** olduğu garanti ediliyor.

Göreviniz, dizideki her elemanı **kırmızı (R)** veya **mavi (B)** renkle boyamaktır. Boyama şu koşulları sağlamalıdır:

1. Kırmızı elemanların aralığı, mavi elemanların aralığına eşit olmamalıdır.  
2. Her iki renkten de en az bir eleman bulunmalıdır.  

Bu koşulları sağlayan bir boyama mümkün değilse, bunu belirtmelisiniz. Eğer birden fazla geçerli boyama varsa, herhangi birini yazabilirsiniz.

## Girdi  
İlk satır bir tamsayı t(1≤t≤100) içerir. — test durumlarının sayısı

Her test durumu için:  
- İlk satır, bir tamsayı n(3≤n≤50) içerir. — dizinin uzunluğu.  
- İkinci satır, sıralı n kadar tamsayı içerir: a₁,a₂,…,aₙ(1≤aᵢ≤10⁹). Bu değerlerin sıralı olarak girilmesi gereklidir: a₁≤a₂≤…≤aₙ₋₁≤aₙ.

## Çıktı  
Her test durumu için:  
- Eğer verilen diziyi boyamak mümkün değilse, **NO** yazdırın.  
- Eğer mümkünse **YES** yazdırın. Ardından n uzunluğunda bir s dizisi yazdırın. 1 ≤ i ≤ n olduğuna göre  sᵢ = R ise aᵢ kırmızı, sᵢ = B ise aᵢ mavidir.

### Girdi
```
7
4
1 1 2 2
5
1 2 3 4 5
3
3 3 3
4
1 2 2 2
3
1 2 2
3
1 1 2
3
1 9 84
```

### Çıktı
```
YES
RBRR
YES
BBRBB
NO
YES
RBBR
YES
RRB
YES
BRR
YES
BRB
```

## Notlar 
1. İlk test durumunda, dizi \([1,1,2,2]\). İkinci eleman mavi, diğerleri kırmızı boyanabilir. Bu durumda, kırmızı elemanların aralığı \([1,2,2]\) için \(2 - 1 = 1\), mavi elemanların aralığı \([1]\) için \(1 - 1 = 0\). Aralıklar farklıdır.  

2. İkinci test durumunda, \([1,2,3,4,5]\) dizisinde ilk, ikinci, dördüncü ve beşinci elemanlar mavi, üçüncü eleman kırmızı boyanabilir. Kırmızı elemanların aralığı \(3 - 3 = 0\) ve mavi elemanların aralığı \(5 - 1 = 4\) olup farklıdır.

3. Üçüncü test durumunda, \([3,3,3]\) dizisini boyamak mümkün değildir çünkü her renk için aralık 0 olacaktır.

**Comment by Burak2210** [comment, marked as answer]
```
t = int(input("t: "))

for x in range(t):
    n = int(input("n: "))
    a = []
    for y in range(n):
        a.append(input("a: "))

    if a[0] == a[n - 1] or n < 3:
        print("NO")
    else:
        print("YES")
        renkler = ""
        for i in range(n):
            if i == 1:
                renkler += "B"
            else:
                renkler += "R"

        print(renkler)
```
Renklerden birini sadece bir sayıya verdim (ilk sayı dışında), yani burada mavinin aralığı her zaman 0 oluyor. Kırmızıların aralığı da bütün değerler aynı olmadığı takdirde her zaman en az 1 oluyor. Yani bütün değerler aynı olmadığı takdirde verilen diziyle istenilen şekilde bir renklendirme yapılabiliyor.

**Reply by Burak2210** [reply]
1 saattir soruyu çözmeye çalışıyordum bi tık geç kaldım hocam

**Reply by sonsuzus** [reply]
Hiç sorun değil geç kalman...
